spark-3/src/test/scala/org/neo4j/spark/DataSourceWriterTSE.scala

```diff
@@ -484,6 +489,175 @@ class DataSourceWriterTSE extends SparkConnectorScalaBaseTSE {
     assertEquals(expected, records)
   }
 
+  private def writeAndGetInterval(expectedDt: Class[_ <: DataType], sql: String): Value = {
+    val id = java.util.UUID.randomUUID().toString
+    val df = sparkSession.sql(s"SELECT '$id' AS id, $sql AS duration")
+
+    df.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Append)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", "Dur")
+      .save()
+
+    val want = expectedDt.getSimpleName
+    val got = df.schema("duration").dataType
+    assertTrue(s"expected Spark to pick $want but it was $got", expectedDt.isInstance(got))
+
+    SparkConnectorScalaSuiteIT.session().run(
+      s"""MATCH (d:Dur {id: '$id'})
+         |RETURN d.duration AS duration
+         |""".stripMargin
+    ).single().get("duration")
+  }
+
+  @Test
+  def `should write nodes with native Neo4j durations when passed DayTimeIntervalType`(): Unit = {
```

Comment:
question: would it be possible to turn all of these test into a single parameterized test? it would make it easier to check whether we've got exhaustive checks on all these interval variants

Reply:
I looked into parametrized tests in sacala/junit4 env, but at that time it was just a few cases. Can look into your suggested intervals and look into re-adding parameterized